cscs-checks/libraries/hpx/hpx_hello_world.py

```diff
@@ -0,0 +1,62 @@
+import itertools
+import os
+
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+@rfm.simple_test
+class HelloWorldHPXCheck(rfm.RunOnlyRegressionTest):
+    def __init__(self):
+        super().__init__()
+
+        self.descr = 'HPX hello, world check'
+        self.valid_systems = ['daint:gpu, daint:mc', 'dom:gpu', 'dom:mc']
+        self.valid_prog_environs = ['PrgEnv-gnu']
+
+        self.modules = ['HPX']
+        self.executable = 'hello_world'
+        self.sourcesdir = None
+
+        self.use_multithreading = None
+
+        self.tags = {'production'}
+        self.maintainers = ['VH', 'JG']
+
+    def setup(self, partition, environ, **job_opts):
+        result = sn.findall(r'hello world from OS-thread \s*(\d+) on '
```

Comment:
I'd call this `hellos`. I'd also give a name in the capturing group, so as to make later the sanity patterns more readable, `tid` and `lid` would make sense.